tests/test_simpleflow/test_step.py

```diff
@@ -54,6 +65,13 @@ def run(self, num, force_steps=[], skip_steps=[]):
                  force_steps_if_executed=['my_step_2']))
         futures.wait(taskf)
 
+    def get_step_bucket(self):
+        return BUCKET
+
+executor = CustomExecutor(MyWorkflow)
```

Comment:
I'd do this a bit differently:
* make `executor` a local variable
* call `create_workflow()` instead of setting `_workflow` directly

Such as:
```diff
diff --git a/tests/test_simpleflow/test_step.py b/tests/test_simpleflow/test_step.py
index de12599..c8b9911 100644
--- a/tests/test_simpleflow/test_step.py
+++ b/tests/test_simpleflow/test_step.py
@@ -35,6 +35,11 @@ class MyTask(object):
 
 
 class CustomExecutor(Executor):
+
+    def __init__(self, workflow_class):
+        super(CustomExecutor, self).__init__(workflow_class)
+        self.create_workflow()
+
     def submit(self, func, *args, **kwargs):
         if hasattr(func, 'activity') and func.activity == MyTask:
             f = futures.Future()
@@ -68,10 +73,6 @@ class MyWorkflow(workflow.Workflow, WorkflowStepMixin):
     def get_step_bucket(self):
         return BUCKET
 
-executor = CustomExecutor(MyWorkflow)
-executor.initialize_history({})
-executor._workflow = MyWorkflow(executor)
-
 
 class StepTestCase(unittest.TestCase, TestWorkflowMixin):
     WORKFLOW = MyWorkflow
@@ -260,6 +261,8 @@ class StepTestCase(unittest.TestCase, TestWorkflowMixin):
         Test that attribute 'raises_on_failure' is well propagated through Step.
         """
         self.create_bucket()
+        executor = CustomExecutor(MyWorkflow)
+        executor.initialize_history({})
 
         activities = Chain(
             (MyTask, 1),
```
(`create_workflow` could be called in `test_propagate_attribute` instead, yep)